### PR TITLE
Respect WP:LEADORDER when tagging

### PR DESCRIPTION
**Pull request by Xi-Plus**
https://github.com/azatoth/twinkle/commit/40592a44443f7b27d44ccc14866ba9526bebf775
morebits.js 和 twinkle.js 已合併到 master branch，Labels 僅作紀錄
friendlytag.js 在 #121 處理

### CODE REVIEW

File: modules/twinkleunlink.js [hunk 1]
```diff
@@ -268,8 +268,7 @@ Twinkle.unlink.callbacks = {
 
 		// remove image usages
 		if (params.doImageusage) {
-			wikiPage.commentOutImage(mw.config.get('wgTitle'), wgULS('注释出', '注釋出'));
-			text = wikiPage.getText();
+			text = wikiPage.commentOutImage(mw.config.get('wgTitle'), wgULS('注释出', '注釋出')).getText();
```
Comment (Hamishcn) [inline modules/twinkleunlink.js:271]: I think we can delete 出 from here

File: modules/twinkleprotect.js [hunk 2]
```diff
@@ -1375,45 +1375,29 @@ Twinkle.protect.callbacks = {
...
-					firstSection = text;
-					otherSection = '';
+			if (/^\s*#(?:redirect|重定向|重新導向)/i.test(text)) { // redirect page
+				// Only tag if no {{rcat shell}} is found
+				if (!text.match(/{{(?:Redirect[ _]category shell|Rcat[ _]shell|This[ _]is a redirect|多种类型重定向|多種類型重定向|多種類型重新導向|多种类型重新导向|R0|其他重定向|RCS|Redirect[ _]shell)/i)) {
+					text = text.replace(/#(?:redirect|重定向|重新導向) ?(\[\[.*?\]\])(.*)/i, '#REDIRECT $1$2\n\n{{' + tag + '}}');
 				} else {
-					firstSection = text.substr(0, index);
-					otherSection = text.substr(index);
+					Morebits.status.info(wgULS('已存在Redirect category shell', '已存在Redirect category shell'), wgULS('没什么可做的', '沒什麼可做的'));
```
Comment (Hamishcn) [inline modules/twinkleprotect.js:1383]: remove nonsense 1st wgULS? I see no translation there

Comment (Xi-Plus) [reply modules/twinkleprotect.js]: Removed

File: modules/twinkleunlink.js (hunk 1 above)
Comment (Xi-Plus) [reply modules/twinkleunlink.js]: Removed

**Pull request merged**